odxtools/diaglayers/ecuvariant.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: MIT
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Any, Dict, Iterable, List, Optional, Union, cast
+from xml.etree import ElementTree
+
+from typing_extensions import override
+
+from ..diagvariable import DiagVariable, HasDiagVariables
+from ..dyndefinedspec import DynDefinedSpec
+from ..ecuvariantpattern import EcuVariantPattern
+from ..exceptions import odxassert
+from ..nameditemlist import NamedItemList
+from ..odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkRef
+from ..parentref import ParentRef
+from ..variablegroup import HasVariableGroups, VariableGroup
+from .diaglayer import DiagLayer
+from .ecuvariantraw import EcuVariantRaw
+from .hierarchyelement import HierarchyElement
+
+
+@dataclass
+class EcuVariant(HierarchyElement):
```

Comment:
Anything that BaseVariant have, should also be available in EcuVariant
```suggestion
class EcuVariant(BaseVariant):
```

Reply:
I agree, but unfortunately this is not how the spec defines things:

![Screenshot_20240805_183755](https://github.com/user-attachments/assets/6613d163-f4bd-4049-9d7a-655e37b10ac6)

Reply:
(note that base variants contain "BASE-VARIANT-PATTERNS" instead of "ECU-VARIANT-PATTERNS".)